src/properties.rs

```diff
@@ -74,15 +99,119 @@ pub enum Property<'a> {
+                variant.newtype_variant()?,
+            )),
+            PropertyIdentifier::ServerKeepAlive => {
+                Ok(Property::ServerKeepAlive(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::AuthenticationMethod => {
+                Ok(Property::AuthenticationMethod(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::AuthenticationData => {
+                Ok(Property::AuthenticationData(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::RequestProblemInformation => Ok(
+                Property::RequestProblemInformation(variant.newtype_variant()?),
+            ),
+            PropertyIdentifier::WillDelayInterval => {
+                Ok(Property::WillDelayInterval(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::RequestResponseInformation => Ok(
+                Property::RequestResponseInformation(variant.newtype_variant()?),
+            ),
+            PropertyIdentifier::ResponseInformation => {
+                Ok(Property::ResponseInformation(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::ServerReference => {
+                Ok(Property::ServerReference(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::ReasonString => {
+                Ok(Property::ReasonString(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::ReceiveMaximum => {
+                Ok(Property::ReceiveMaximum(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::TopicAliasMaximum => {
+                Ok(Property::TopicAliasMaximum(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::TopicAlias => Ok(Property::TopicAlias(variant.newtype_variant()?)),
+            PropertyIdentifier::MaximumQoS => Ok(Property::MaximumQoS(variant.newtype_variant()?)),
+            PropertyIdentifier::RetainAvailable => {
+                Ok(Property::RetainAvailable(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::UserProperty => {
+                // TODO:
+                //Ok(variant.tuple_variant(UserPropertyVisitor)?)
+                Err(A::Error::custom("User property not yet implemented"))
+            }
+            PropertyIdentifier::MaximumPacketSize => {
+                Ok(Property::MaximumPacketSize(variant.newtype_variant()?))
+            }
+            PropertyIdentifier::WildcardSubscriptionAvailable => Ok(
+                Property::WildcardSubscriptionAvailable(variant.newtype_variant()?),
+            ),
+            PropertyIdentifier::SubscriptionIdentifierAvailable => Ok(
+                Property::SubscriptionIdentifierAvailable(variant.newtype_variant()?),
+            ),
+            PropertyIdentifier::SharedSubscriptionAvailable => Ok(
+                Property::SharedSubscriptionAvailable(variant.newtype_variant()?),
```

Comment:
There should be some simple macros (map?) that can convert all the PropertyIdentifiers to `Property::<Type>(variant.newtype_variant()?)` to kill off a lot of this boilerplate. The `UserProperty` is a special case since it's a tuple_variant